cpp/ql/src/semmle/code/cpp/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -173,6 +174,12 @@ private predicate exprToExprStep(Expr exprIn, Expr exprOut) {
       exprIn = call.getQualifier()
     )
   )
+  or
+  exists(Variable iterator, Variable collection |
+    assignmentViaIterator(iterator, exprIn) and
+    isIteratorForCollection(iterator, collection) and
+    collection.getAnAccess() = exprOut
```

Comment:
This is certainly a step in the right direction but carries a risk of false positives and quadratic performance. Some tests showing false positives would be helpful when evaluating improvements to this code.